src/NHibernate/Criterion/ProjectionsExtensions.cs

```diff
@@ -323,5 +334,20 @@ internal static IProjection ProcessMod(MethodCallExpression methodCallExpression
 			object divisor = ExpressionProcessor.FindValue(methodCallExpression.Arguments[1]);
 			return Projections.SqlFunction("mod", NHibernateUtil.Int32, property, Projections.Constant(divisor));
 		}
+
+		/// <summary>
+		/// Project Entity
+		/// </summary>
+		public static T AsEntity<T>(this T alias) where T:class
+		{
+			throw new Exception("Not to be used directly - use inside QueryOver expression");
```

Comment:
Do not throw bare `Exception`. For such case we tend to throw an `InvalidOperationException` (as in `MappedAs` Linq function).

Reply:
I just copy-pasted it from some method above. Actually all methods in this `ProjectionsExtensions `class throw this `Exception("Not to be used directly...`. So I guess it's better be uniform and throw the same kind of exception for all cases. And if change it - it should be done for all cases at once and not in scope of this PR. Don't you agree?

Reply:
Ok, I was thinking the code was already purged from throwing bare `Exception`, but in fact that is only `ApplicationException` which were removed (#602, its title applies to the cases where it served as base class).

So yes, keep it that way.

Reply:
Need to be aligned with #1498